### PR TITLE
Adding in the basic restructuring of intro-to-windows-containers.

### PR DESCRIPTION
**Pull request by brasmith-ms**
Creating the initial restructuring of the existing Windows kubernetes documentation into more management categories. Categories are somewhat arbitrary so feel free to offer different suggestions.

Per #31428

### CODE REVIEW

File: content/en/docs/setup/windows/intro-windows-in-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+  * A UI thread handles well-defined messages including `WM_CLOSE`.
+  * Console apps handle Ctrl-C or Ctrl-break using a Control Handler.
+  * Services register a Service Control Handler function that can accept
+    `SERVICE_CONTROL_STOP` control codes.
+
+Container exit codes follow the same convention where 0 is success, and nonzero is failure.
+The specific error codes may differ across Windows and Linux. However, exit codes
+passed from the Kubernetes components (kubelet, kube-proxy) are unchanged.
+
+##### Field compatibility for container specifications {#compatibility-v1-pod-spec-containers}
```
Comment (sftim) [inline content/en/docs/setup/windows/intro-windows-in-kubernetes.md:187]: The level 5 headings here should, I think, become level 4 headings.

Ideally: move these field compatibility details into a new Reference page, and add inbound hyperlinks from relevant pages.

File: content/en/docs/setup/windows/intro-windows-in-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+
+### Node problem detector
+
+The node problem detector (see
+[Monitor Node Health](/docs/tasks/debug-application-cluster/monitor-node-health/))
+is not compatible with Windows.
+
+### Pause container
+
+In a Kubernetes Pod, an infrastructure or “pause” container is first created
```
Comment (sftim) [inline content/en/docs/setup/windows/intro-windows-in-kubernetes.md:267]: nit (not related to the refactor):
```suggestion
In a Kubernetes Pod, an infrastructure or _pause_ container is first created
```

I'll file an issue to add “pause container” to the glossary.

File: content/en/docs/setup/windows/intro-windows-in-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+
+Microsoft maintains a different multi-architecture image, with Linux and Windows
+amd64 support, that you can find as `mcr.microsoft.com/oss/kubernetes/pause:3.6`.
+This image is built from the same source as the Kubernetes maintained image but
+all of the Windows binaries are [authenticode signed](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/authenticode) by Microsoft.
+The Kubernetes project recommends using the Microsoft maintained image if you are
+deploying to a production or production-like environment that requires signed
+binaries.
+
+### Container runtimes {#container-runtime}
```
Comment (sftim) [inline content/en/docs/setup/windows/intro-windows-in-kubernetes.md:288]: I'd make this a second level heading, and adjust the headings below it to match.

File: content/en/docs/setup/windows/intro-windows-in-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+* [Troubleshooting Windows Networking](/docs/setup/windows/windows-networking#troubleshooting-network)
+* [Node-level Troubleshooting](/docs/setup/windows/user-guide-windows-containers#troubleshooting-node)
+
+### Further investigation
+
+If these steps don't resolve your problem, you can get help running Windows containers on Windows nodes in Kubernetes through:
+
+* StackOverflow [Windows Server Container](https://stackoverflow.com/questions/tagged/windows-server-container) topic
+* Kubernetes Official Forum [discuss.kubernetes.io](https://discuss.kubernetes.io/)
+* Kubernetes Slack [#SIG-Windows Channel](https://kubernetes.slack.com/messages/sig-windows)
```
Comment (sftim) [inline content/en/docs/setup/windows/intro-windows-in-kubernetes.md:358]: Let's omit this from the overview page - we should instead add that to all of the Windows-specific troubleshooting pages.

(As a later piece of work, anyone who's interested in cleanup can refactor it into a reusable shortcode).

File: content/en/docs/setup/windows/intro-windows-in-kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+the issue to a SIG Windows member's attention
+
+
+## {{% heading "whatsnext" %}}
+
+### Deployment tools
+
+The kubeadm tool helps you to deploy a Kubernetes cluster, providing the control
+plane to manage the cluster it, and nodes to run your workloads.
+[Adding Windows nodes](/docs/tasks/administer-cluster/kubeadm/adding-windows-nodes/)
```
Comment (sftim) [inline content/en/docs/setup/windows/intro-windows-in-kubernetes.md:389]: Does this link need to change? “Adding Windows nodes” should probably move into the Setup section.

File: content/en/docs/setup/windows/windows-networking.md [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+
+* Windows reference network plugins win-bridge and win-overlay do not implement
+  [CNI spec](https://github.com/containernetworking/cni/blob/master/SPEC.md) v0.4.0,
+  due to a missing `CHECK` implementation.
+* The Flannel VXLAN CNI plugin has the following limitations on Windows:
+
+1. Node-pod connectivity isn't possible by design. It's only possible for local pods with Flannel v0.12.0 (or higher).
+2. Flannel is restricted to using VNI 4096 and UDP port 4789. See the official
+   [Flannel VXLAN](https://github.com/coreos/flannel/blob/master/Documentation/backends.md#vxlan)
+   backend docs for more details on these parameters.
```
Comment (sftim) [inline content/en/docs/setup/windows/windows-networking.md:115]: nit (not related to refactor): I think these lines should be indented by two spaces
```suggestion
  1. Node-pod connectivity isn't possible by design. It's only possible for local pods
     with Flannel v0.12.0 (or higher).
  2. Flannel is restricted to using VNI 4096 and UDP port 4789. See the official
     [Flannel VXLAN](https://github.com/coreos/flannel/blob/master/Documentation/backends.md#vxlan)
     backend docs for more details on these parameters.
```

File: content/en/docs/setup/windows/windows-networking.md [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+| Preserve-Destination | Skips DNAT of service traffic, thereby preserving the virtual IP of the target service in packets reaching the backend Pod. Also disables node-node forwarding. | v1.20+ | Windows Server, version 1903 (or higher) | Set `"preserve-destination": "true"` in service annotations and enable DSR in kube-proxy. |
+| IPv4/IPv6 dual-stack networking | Native IPv4-to-IPv4 in parallel with IPv6-to-IPv6 communications to, from, and within a cluster | v1.19+ | Windows Server, version 2019 | See [IPv4/IPv6 dual-stack](#ipv4ipv6-dual-stack) |
+| Client IP preservation | Ensures that source IP of incoming ingress traffic gets preserved. Also disables node-node forwarding. | v1.20+ | Windows Server, version 2019  | Set `service.spec.externalTrafficPolicy` to "Local" and enable DSR in kube-proxy |
+{{< /table >}}
+
+### Session affinity
+
+Setting the maximum session sticky time for Windows services using
+`service.spec.sessionAffinityConfig.clientIP.timeoutSeconds` is not supported.
```
Comment (sftim) [inline content/en/docs/setup/windows/windows-networking.md:156]: I'd move this into https://kubernetes.io/docs/concepts/services-networking/service/#proxy-mode-ipvs, inside a [note shortcode](https://kubernetes.io/docs/contribute/style/style-guide/#note).

File: content/en/docs/setup/windows/windows-networking.md [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+{{< note >}}
+Overlay (VXLAN) networks on Windows do not support dual-stack networking.
+{{< /note >}}
+
+# Network troubleshooting {#troubleshooting-network}
+
+1. My Windows Pods do not have network connectivity
+
+   If you are using virtual machines, ensure that MAC spoofing is **enabled** on all
+   the VM network adapter(s).
```
Comment (sftim) [inline content/en/docs/setup/windows/windows-networking.md:191]: We should also make this clear somewhere in the setup section (maybe we already do?)

File: content/en/docs/setup/windows/windows-networking.md [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+   Often it is worthwhile to modify the [InterfaceName](https://github.com/microsoft/SDN/blob/master/Kubernetes/flannel/start.ps1#L7) parameter of the start.ps1 script, in cases where the host's network adapter isn't "Ethernet". Otherwise, consult the output of the `start-kubelet.ps1` script to see if there are errors during virtual network creation.
+
+1. DNS resolution is not properly working
+
+   Check the DNS limitations for Windows in this [section](#dns-limitations).
+
+1. `kubectl port-forward` fails with "unable to do port forwarding: wincat not found"
+
+   This was implemented in Kubernetes 1.15 by including `wincat.exe` in the pause infrastructure container `mcr.microsoft.com/oss/kubernetes/pause:3.6`. Be sure to use a supported version of Kubernetes.
+   If you would like to build your own pause infrastructure container be sure to include [wincat](https://github.com/kubernetes/kubernetes/tree/master/build/pause/windows/wincat).
```
Comment (sftim) [inline content/en/docs/setup/windows/windows-networking.md:296]: nit: is this out of date?

**Review by sftim — COMMENTED** [review]
:tada: This is a really welcome PR @brasmith-ms !

My thoughts at this point (I can see it's still a draft):

- `content/en/docs/setup/windows/user-guide-windows-containers.md` is _not_ a cluster setup topic.
   - once you have an existing cluster that supports Windows pods, running a new Windows pod is a reasonable thing to want to do at any point


You can (and should) redirect people from the old pages to the new ones. Edit `static/_redirects` to do that.

/label refactor

There is also a bunch of inline feedback, that I hope you do find useful.

**Comment by marosset** [comment]
I think that it might still be good to have one spot act as a 'landing page' for Windows support in Kubernetes.
We can use that page to hold some general information and also link to the different pages.
Would it make sense create a new page/folder under https://kubernetes.io/docs/reference/ to act as this landing page?
I do not think it belongs under https://kubernetes.io/docs/setup/production-environment/windows/
@sftim Do you have any thoughts on this?

**Comment by tengqm** [comment]
@brasmith-ms There are conflicts to be resolved ...

**Comment by marosset** [comment]
/close
We are going to break this up into multiple pull-requests to make it easier to review / avoid merge conflicts.
Thanks!
@brasmith-ms  FYI

**Pull request closed without merging**